server/partition.go

```diff
@@ -825,6 +832,25 @@ func (p *partition) messageProcessingLoop(recvChan <-chan *nats.Msg, stop <-chan
 		// Write uncommitted messages to log.
 		offsets, err := p.log.Append(msgBatch)
 		if err != nil {
+
+			// If ErrIncorrectOffset is raised, then
+			// and AckErr should be dispatched
+
+			if err == commitlog.ErrIncorrectOffset {
```

Comment:
```suggestion
			if errors.Is(err, commitlog.ErrIncorrectOffset) {
```

Liftbridge enforces Go 1.14 in the go.mod so we should start using `errors.Is` and `errors.As`. /cc @tylertreat WDYT?